src/components/__tests__/StateResults.js

```diff
@@ -21,33 +24,87 @@ it("doesn't render if no results", () => {
   expect(wrapper.html()).toBeUndefined();
 });
 
-it('gives results to default slot', () => {
+it('gives state & results to default slot', () => {
   const results = {
     query: 'q',
     hits: [{ objectID: '1', name: 'one' }, { objectID: '2', name: 'two' }],
     page: 1,
   };
+  const state = {
+    query: 'something',
+    disjunctiveFacetsRefinements: {},
+    page: 1,
+  };
 
   __setState({
+    state,
     results,
   });
 
   mount(StateResults, {
     scopedSlots: {
-      default: props => expect(props).toEqual(results),
+      default: props => {
+        expect(props).toEqual(expect.objectContaining(results));
+        expect(props.results).toEqual(results);
+        expect(props.state).toEqual(state);
+      },
     },
   });
 });
 
-it('allows default slot to render whatever they want (truthy query)', () => {
+it('allows default slot to render whatever they want (modern scope)', () => {
```

Comment:
We could have kept the tests that were present before but scope them under `describe('legacy usage')`. Then create new ones with the same state for the new usage.